crates/driver/src/tests/setup/solver.rs

```diff
@@ -0,0 +1,62 @@
+use {
+    std::{net::SocketAddr, sync::Arc},
+    tokio::sync::Mutex,
+};
+
+/// Configuration for mocking a solver.
+#[derive(Debug, Default)]
+pub struct Config {
+    /// The auctions and solutions that the solver should expect and respond
+    /// with.
+    pub solve: Vec<Solve>,
+    pub name: String,
+    pub absolute_slippage: String,
+    pub relative_slippage: String,
+    pub address: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct Solve {
+    pub req: serde_json::Value,
+    pub res: serde_json::Value,
+}
+
+/// A solver that was started as part of the setup.
+#[derive(Debug)]
+pub struct Solver {
+    pub config: Config,
+    pub addr: SocketAddr,
+}
+
+/// Set up an HTTP server exposing a solver API and acting as a solver mock.
+pub async fn setup(config: Config) -> Solver {
+    let state = Arc::new(Mutex::new(config.solve.clone()));
+    let app = axum::Router::new()
+        .route(
+            "/",
+            axum::routing::post(
+                |axum::extract::State(state): axum::extract::State<State>,
+                 axum::extract::Json(req): axum::extract::Json<serde_json::Value>| async move {
+                    let mut state = state.0.lock().await;
```

Comment:
An async aware `Mutex` is only needed if you want to hold the lock across `.await`s.

Reply:
Purely technical side:
You definitely shouldn't use a sync mutex across an await point. When there is no await point it depends on how long one expects acquiring the lock to usually take. If you're just incrementing an integer or cloning struct then a sync mutex is going to be faster. When you're waiting for some long expensive computation then it should be an async mutex.

Non technical:
One reason I prefer the sync mutex if possible is that it is a simpler component. An async mutex is a sync mutex with extra stuff. I don't want to involve extra stuff when it's not needed.  Another reason is that it communicates the intent to some degree. If I see a sync mutex in async code then I know it's meant to be held for short durations.

It's true that there is some danger of forgetting what kind of mutex it is but usually I still prefer the sync mutex even when taking that into account. Often it is clear that the operation is fast and there is little danger of misusing the mutex.

Vaguely related, I also wouldn't mark a function that doesn't need to be async as async just in case. It's more efficient to do this but it also feels more correct. For an async function there is no danger of misuse because the compiler will help you so the downside is smaller. But, I think clippy already has the equivalent warning for a sync Mutex held across await point.